src/services/ActivityService.ts

```diff
@@ -84,39 +84,37 @@ export class ActivityService {
 
     /**
      * Ends an activity with the given id
+     * if the visit was already closed, the response will be 200 and the flag visitClosed will be set to true
      * @param id - id of the activity to end
-     * @returns Promise void
+     * @returns Promise<{visitClosed: boolean}>
      */
-    public async endActivity(id: string): Promise<void> {
-        return this.dbClient.get({ id })
-            .then(async (result: DocumentClient.GetItemOutput): Promise<void> => {
-                // Result checks
-                if (result.Item === undefined) {
-                    throw new HTTPResponse(404, { error: HTTPRESPONSE.NOT_EXIST });
-                }
-
-                if (result.Item.endTime !== null) {
-                    throw new HTTPResponse(403, { error: HTTPRESPONSE.ALREADY_ENDED });
-                }
+    public async endActivity(id: string): Promise<{ visitClosed: boolean }> {
+        try {
+            const result: DocumentClient.GetItemOutput = await this.dbClient.get({ id });
+            if (result.Item === undefined) {
+                throw new HTTPResponse(404, { error: HTTPRESPONSE.NOT_EXIST });
+            }
 
+            if (result.Item.endTime !== null) {
+                console.log(`Visit with ID ${id} is already closed`);
+                return { visitClosed: true };
+            } else {
```

Comment:
else would not be needed as if has a return. suggest to remove else.

Reply:
done